bigbite/themer
# [BBT-93]: Outline UI component

## Description



Completes [BBT-93](https://b5ecom.atlassian.net/browse/BBT-93) - This PR adds a reusable outline component to the plugin. The component allows you to set 4 outline properties:

- Width
- Offset
- Colour
- Style

Similar to #22, I've limited the available units you can use for width & offset as some commonly used units aren't supported (e.g. %). If an unsupported unit is used, it defaults to `px`.

## Change Log



- Added shadow component to block styles section of the plugin.

## Steps to test



- Add some outline styles to `theme.json` for one of the blocks. For example:
```json
"styles": {
  "blocks": {
    "core/paragraph": {
      "outline": {
        "offset": "3px",
        "width": "3px",
        "style": "dashed",
        "color": "var(--wp--preset--color--secondary)"
        }
    }
}
```
- Check that the values you added are imported correctly by the component
- Change some values & check that your changes are accurately reflected in the preview
- After you've made some changes, click save & then export `theme.json`. Check your changes have been exported correctly into the new `theme.json`. 


## Screenshots/Videos



### Importing from `theme.json`

https://github.com/bigbite/themer/assets/45216035/4e2dc49d-be1b-436c-a321-0a4aed5b29ed

### Changing values

https://github.com/bigbite/themer/assets/45216035/14af22e7-10b6-4eec-9662-b86644b3e757

### Exporting changes

https://github.com/bigbite/themer/assets/45216035/98094b1c-5c48-4cc6-8f31-9d636964b374

## Checklist:





[BBT-93]: https://b5ecom.atlassian.net/browse/BBT-93?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
